port/unix/omrsysinfo.c

```diff
@@ -3459,38 +3542,68 @@ omrsysinfo_cgroup_is_limits_supported(struct OMRPortLibrary *portLibrary)
 			goto _end;
 		}
 
+		omrthread_monitor_enter(cgroupEntryListMonitor);
 		rc = readCgroupFile(portLibrary, getpid(), &PPG_cgroupEntryList);
+		omrthread_monitor_exit(cgroupEntryListMonitor);
+		if (0 != rc) {
+			goto _end;
+		}
+		PPG_cgroupSubsystemsAvailable = getAvailableSubsystems(portLibrary, PPG_cgroupEntryList);
 	} else {
 		rc = 0;
 	}
-
 _end:
 #endif /* defined(LINUX) */
 	return rc;
 }
 
-BOOLEAN 
-omrsysinfo_cgroup_is_limits_enabled(struct OMRPortLibrary *portLibrary)
+uint64_t
+omrsysinfo_cgroup_get_available_subsystems(struct OMRPortLibrary *portLibrary)
+{
+#if defined(LINUX)
+	if (NULL == PPG_cgroupEntryList) {
+		int32_t rc = portLibrary->sysinfo_cgroup_is_system_available(portLibrary);
+		if (0 != rc) {
+			PPG_cgroupSubsystemsAvailable = 0;
```

Comment:
I don't like the way this global is sometimes set here, sometimes set in sysinfo_cgroup_is_system_available.  It needs to be set in a single place (single point of initialization)

Reply:
I moved it to `omrsysinfo_cgroup_is_system_available`.